shared/src/metabase/mbql/util.cljc

```diff
@@ -278,6 +278,19 @@
     [(op :guard temporal-extract-ops) field & args]
     [:temporal-extract field (temporal-extract-ops->unit [op (first args)])]))
 
+(defn desugar-convert-timezone
+  "Replace the magic argument of convert-timezone expression.
+
+  Currently only replace \"Instance\" with `results-timezone-id`."
+  [m results-timezone-id]
+  (mbql.match/replace m
+    [(op :guard #{:convert-timezone}) field & args]
+    (into [:convert-timezone field ] (map (fn [arg]
+                                            (if (= (str/lower-case arg) "instance")
+                                             results-timezone-id
+                                             arg))
+                                          args))))
+
 (s/defn desugar-filter-clause :- mbql.s/Filter
```

Comment:
`desugar-convert-timezone` needs to be part of this too in order to desugar convert-timezone in custom filter expressions

Reply:
added here ce7d15314c